packages/components/src/Disclosure/Disclosure.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { ReactNode, useState } from "react";
+import classnames from "classnames";
+import styles from "./Disclosure.css";
+import { Heading } from "../Heading";
+import { Icon } from "../Icon";
+
+export interface DisclosureProps {
+  /**
+   * Child content that is manged by this component.
+   */
+  readonly children: ReactNode | ReactNode[];
+
+  /**
+   * Title for the disclosure pane.
+   */
+  readonly title: string;
+
+  /**
+   * Control if the disclosure component is open or closed.
+   */
+  readonly open?: boolean;
+
+  /**
+   * This sets the default open state of the disclosure.
+   * By default the disclosure is closed.
+   * For use when the component is being used as an Uncontrolled Component.
+   * @default false
+   */
+  readonly defaultOpen?: boolean;
+
+  /**
+   * This function would toggle the open state.
+   * For use when the component is being used as a Controlled Component.
+   */
+  onRequestToggle?(isOpen: boolean): void;
```

Comment:
I think it might be good to change the verbiage on the parameter here as it's really requesting this new state.

Perhaps, `requestOpen`, `shouldOpen`?

I was also wondering if two methods a `onRequestOpen` and `onRequestClose` would be better but, I don't think I like that if we can get a good name for the parameter.